ci/Jenkinsfile-pyp

```diff
@@ -0,0 +1,205 @@
+pipeline {
+    agent any
+
+    parameters {
+        choice(
+            name: 'release_type',
+            choices: ['dev', 'stable'],
+            description: 'Indicate whether a dev or stable release should be built.'
+        )
+        choice(
+            name: 'version_increment',
+            choices: ['patch', 'minor', 'major'],
+            description: 'Indicates what kind of version bump should be done'
+        )
+        booleanParam(name: 'run_job', defaultValue: false, description: 'Allow for dry-runs')
```

Comment:
This could do with a bit more explanation of what is done/not done when this box is ticked